tests/ndarrays/test_ndarrays.c

```diff
@@ -0,0 +1,278 @@
+
+#define my_assert(X , Y, dscr, __func__, __FILE__, __LINE__) _Generic((X), double: assert_double,\
+							float: assert_float,\
+							int: assert_int,\
+							double complex : assert_complex_double,\
+							default: assert_ns)(X , Y, dscr, __func__, __FILE__, __LINE__)
+#define getname(X) #X
+
+void assert_double(double v1 , double v2, const char *dscr, const char * func, const char *file, int line)
+{
+
+	if (v1 != v2)
+	{
+		printf("[FAIL] %s:%d:%s\n", file, line, func);
+		printf("[INFO] %s:%f != %s:%f\n", getname(v1), v1, getname(v2),v2);
+		printf("[DSCR] %s\n", dscr);
+		return ;
+	}
+	printf("[PASS] %s:%d:%s\n", file, line, func);
+	printf("[DSCR] %s\n", dscr);
+}
+
+void assert_float(float v1 , float v2, const char *dscr, const char * func, const char *file, int line)
+{
+	if (v1 != v2)
+	{
+		printf("[FAIL] %s:%d:%s\n", file, line, func);
+		printf("[INFO] %s:%f != %s:%f\n", getname(v1), v1, getname(v2),v2);
+		printf("[DSCR] %s\n", dscr);
+		return ;
+	}
+	printf("[PASS] %s:%d:%s\n", file, line, func);
+	printf("[DSCR] %s\n", dscr);
+}
+
+void assert_int(int v1 , int v2, const char *dscr, const char * func, const char *file, int line)
+{
+	if (v1 != v2)
+	{
+		printf("[FAIL] %s:%d:%s\n", file, line, func);
+		printf("[INFO] %s:%d != %s:%d\n", getname(v1), v1, getname(v2),v2);
+		printf("[DSCR] %s\n", dscr);
+		return ;
+	}
+	printf("[PASS] %s:%d:%s\n", file, line, func);
+	printf("[DSCR] %s\n", dscr);
+}
+
+void assert_complex_double(double complex v1 , double complex v2, const char *dscr, const char * func, const char *file, int line)
+{
+	if (v1 != v2)
+	{
+		printf("[FAIL] %s:%d:%s\n", file, line, func);
+		printf("[INFO] %s:%f+%f*I != %s:%f+%f*I\n", getname(v1), creal(v1), cimag(v1), getname(v2),creal(v2), cimag(v2));
+		printf("[DSCR] %s\n", dscr);
+		return ;
+	}
+	printf("[PASS] %s:%d:%s\n", file, line, func);
+	printf("[DSCR] %s\n", dscr);
+}
+void assert_ns(float v1 , float v2, const char *dscr, const char * func, const char *file, int line)
+{
+
+	printf("[FAIL] %s:%d:%s\n", file, line, func);
+	printf("[INFO] not supported type\n");
+	printf("[DSCR] %s\n", dscr);
+	return ;
+}
+
+int	test_indexing_int(void)
+{
+	int m_1[] = {2, 3, 5, 5, 6, 7, 10, 11, 12, 260, 6, 8, 8, 0, 45, 0, 1, 0, 0, 1, 200, 33, 5, 57, 62, 70, 103, 141, 122, 26, 36, 82, 8, 10, 4115, 22, 1, 11, 1, 19};
+	int m_1_shape[] = {5, 8};
+	t_ndarray *x;
+	int index;
+	int c_index;;
+	int value;
+	int c_value;;
+
+	x = init_array((char *)m_1, 2, m_1_shape, sizeof(int));
+	// testing the index [3, 2]
+	index = 3 * x->strides[0] + 2 * x->strides[1];
+	c_index = 26;
+	m_assert(index , c_index, "testing the strides"); //testing the strides
+	m_assert(get_index(x, 3, 2) , c_index, "testing the indexing function"); //testing the indexing function
+	// testing the value with the index [0, 0]
+	value = x->data->int_nd[index];
+	c_value = 103;
+	m_assert(value , c_value, "testing the value");
+	free_array(x);
+	return (1);

```

Comment:
Why return 1? Is this to signify true? If so why not use a boolean? Or to signify no problems? Why not 0, the standard error code that you have used in the main?
Although I am not sure that it makes sense to return anything. This doesn't even indicate that the assertions were successful

Reply:
i will remove it && and will use 0 in the future to indicate true

Reply:
0 to indicate no errors ^^
0 to indicate true is just confusing

Reply:
ah okey make sens